src/frontend/src/optimizer/plan_node/generic.rs

```diff
@@ -845,26 +845,25 @@ pub struct TopN<PlanRef> {
     pub group_key: Vec<usize>,
 }
 
-pub trait GenericBase {
-    fn schema(&self) -> &Schema;
-    fn logical_pk(&self) -> &[usize];
+pub trait GenericPlanNode {
+    fn schema(&self) -> Schema;
+    fn logical_pk(&self) -> Vec<usize>;
```

Comment:
Are you sure about the copy?

Reply:
yes! you can see the project's implementation, before a plannode(the inner struct) generating plan base and converting to plan ref, we can not get a reference of these property easily